### PR TITLE
added change-information, change-password endpoints. refactored manua…

### PR DESCRIPTION
**Pull request by dylannandlall**
added change-information, change-password endpoints. refactored manual_register function in auth.py. Created new login function for manual login within auth.py. Added update_user, find_user_by_email, and verify_password database functions within util.py

### CODE REVIEW

File: backend/routes/auth.py [hunk 1]
```diff
@@ -44,3 +44,23 @@ def manual_register():
...
+        email = request.form['email']
+        password = request.form['password']
+    
+        if find_user_by_email(email):
+            if verify_password(email, password):
+                return Response("Login Successful", status=200, mimetype='application/json')
+            else:
+                return Response("Login Failed", status=401, mimetype='application/json')
+        else:
+            return Response("Login Failed", status=401, mimetype='application/json')
```
Comment (KyoshiNoda) [inline backend/routes/auth.py]: **_MANDATORY:_**
- Throw errors for each condition. 
- At the end, send a Response OK

File: backend/routes/users.py [hunk 2]
```diff
@@ -8,3 +10,43 @@ def get_users():
...
     users_list = [user.to_dict() for user in users]
     return Response(json.dumps(users_list), status=200, content_type='application/json')
+
+@users_blueprint.route("change-information", methods=["PATCH"])
+def change_user_information():
+    """
+    email: str
+    password: str
+    steamurl: str
+    """
```
Comment (KyoshiNoda) [inline backend/routes/users.py]: remove comments

File: backend/routes/users.py [hunk 3]
```diff
@@ -8,3 +10,43 @@ def get_users():
...
+        return Response(status=200, response=json.dumps({"message": "User information changed successfully"}), content_type='application/json')
+    else:
+        return Response(status=400, response=json.dumps({"error": "User information not changed"}), content_type='application/json')
+    
+@users_blueprint.route("change-password", methods=["PATCH"])
+def change_user_password():
+    """
+    email: str
+    password: str
+    """
```
Comment (KyoshiNoda) [inline backend/routes/users.py]: remove comments

File: backend/routes/auth.py [hunk 4]
```diff
@@ -3,16 +3,16 @@
 from steam.api import get_steam_id, get_player_summary
 import bcrypt
 auth_blueprint = Blueprint('auth', __name__)
-
+from database.util import find_user_by_email, verify_password
```
Comment (KyoshiNoda) [inline backend/routes/auth.py]: consider doing `*` for all future imports
```suggestion
from database.util import *
```

File: backend/database/util.py [hunk 5]
```diff
@@ -19,7 +20,30 @@ def create_user(userdata: dict):
...
+        print(e)
+        return False 
 
 def find_user(steamid):
     user = User.query.filter_by(steamid=steamid).first()
     return bool(user)
+
+def find_user_by_email(email):
+    user = User.query.filter_by(email=email).first()
+    return bool(user)
```
Comment (KyoshiNoda) [inline backend/database/util.py:45]: **_MANDATORY:_** return the `user.to_dict()`
```suggestion
return user.to_dict() if user else None
```

File: backend/database/util.py [hunk 6]
```diff
@@ -19,7 +20,30 @@ def create_user(userdata: dict):
...
+        user = User.query.filter_by(email=email).first()
+        if user:
+            for key, value in userdata.items():
+                setattr(user, key, value)
+        db.session.commit()
+        return True
+    except Exception as e:
+        db.session.rollback()
+        print(e)
+        return False
```
Comment (KyoshiNoda) [inline backend/database/util.py]: **_MANDATORY:_**
I rather send the user itself everytime we update. I got it working on my branch. Could you just use the following instead? I have test already.
```suggestion
def update_user(steam_id, new_fields):
    try:
        user = User.query.filter_by(steam_id=steam_id).first()
        if user:
            for key, value in new_fields.items():
                setattr(user, key, value)
            db.session.commit()
            return user.to_dict()
        else:
            return None
    except Exception as e:
        db.session.rollback()
        return e
```

**Review by KyoshiNoda — CHANGES_REQUESTED** [review]
looks really good! 🐐

Requested 3 changes, the rest are optionals/suggestions.

**Review by dylannandlall — COMMENTED** [review]
looks good

**Review by dylannandlall — COMMENTED** [review]
changes made, inform if they are acceptable.

**Review by KyoshiNoda — APPROVED** [review]

**Pull request merged**